components/logins/src/sync/engine.rs

```diff
@@ -950,4 +963,111 @@ mod tests {
             "else"
         );
     }
+
+    fn count(engine: &LoginsSyncEngine, sql: &str) -> u32 {
```

Comment:
nit: What do you think of having this function take the table name to count, instead of the whole SQL statement?